Ma2694_Assignment 2.Rmd

```diff
@@ -0,0 +1,195 @@
+    ````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+
+#countin teacher activity
+teacher.count<- EDGE %>% group_by(from) %>% summarise(t.actv=sum(count))
+names(teacher.count) <- c("to", "count")
+
+#keeping relevant variables from student data frame
+EDGE2<-select(EDGE,-1,-4)
+
+#combining student and teacher dataframe
+EDGE3<-bind_rows(EDGE2,teacher.count)
+
+#changing name
+setnames(EDGE3,"to","id")
+
+#creating a new vertex
+VERTEX2<-full_join(VERTEX,EDGE3,by="id" )
+
+#plotting
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, vertex.size=VERTEX2$count)
+```
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+
+```{r}
+library(dplyr)
+library(tidyr)
+library(igraph)
+
+#laoding data
+class<-read.csv("HUDK4050-2018-classes.csv", header=TRUE, sep = ",")
+
+#removing odd rows
+class2<-class[-1:-2,]
+class2<-class2[-37,]
+
+#keeping variables
+class2<-select(class2,15:21)
+
+#gathering classes
+class3<-gather(class2,"class","name",3:7)
+
+#forming an unique student name
+class3$student<-paste(class3[,1],class3[,2])
+
+#removing empty classes
+class4<-class3[class3$name!="",]
+
+#keeping variables
+class4<-select(class4,4:5)
+
+#re-ordering columns, student first
+class4 <- class4[c(2,1)]
+
+#there are repeated courses written differently, thus I'm making them unique
+class4$name <- gsub(" ", "", class4$name)
+
+#since one student did it twice
+class5<-unique(class4)
+
+#creating a counter
+class5$counter<-1
+
+#making data frame person to person
+class5<-spread(class5,2,3,fill = 0)
+
+#removing student name
+class6<-select(class5,-1)
+
+#adding student name inner table
+row.names(class6) <-class5[,1]
+
+# as matrix
+## person-class
+p.c<-as.matrix(class6)
+
+## person-person
+p.p<-p.c %*% t(p.c)
+
+## graph
+g2 <- graph_from_adjacency_matrix(p.p, mode = c("undirected"), weighted = NULL, diag = FALSE,
+  add.colnames = NULL, add.rownames = NA)
+plot(g2)
+
+#betweenness
+
+degree(g2)  #degree
+betweenness(g2, v = V(g2), directed = TRUE, weights = NULL,nobigint = TRUE, normalized = FALSE)
+
+#Xixuan Zhao has the greatest number of links and nodes with other students in the class, becoming him Xixuan the most central person in the class. 
+
```

Comment:
I appreciate you providing some comments about the most "influential" person in the class. Going forward, can you think of ways to programmatically pull out this person? Good job!